frontend/src/pages/PipelineDetailsV1.test.tsx

```diff
@@ -121,10 +130,131 @@ spec:
     expect(screen.getByTestId('version_selector').childElementCount).toEqual(1);
   });
 
-  it('shows description for pipeline version', async () => {
+  it('shows description for pipeline version and pipeline with custom version', async () => {
+    render(
+      <PipelineDetailsV1
+        {...generateProps(
+          new graphlib.Graph(),
+          new graphlib.Graph(),
+          'test-pipeline-version-desc',
+          'test-pipeline-desc',
+          'test-default-version-desc',
+        )}
+      />,
+    );
+
+    expect(screen.getByText('test-pipeline-desc'));
+    expect(screen.getByText('test-pipeline-version-desc'));
+    expect(screen.queryByText('Default Version Description')).toBeNull();
+    expect(screen.queryByText('test-default-version-desc')).toBeNull();
+  });
+
+  it('shows description for pipeline version and pipeline with default version', async () => {
+    render(
+      <PipelineDetailsV1
+        {...generateProps(
+          new graphlib.Graph(),
+          new graphlib.Graph(),
+          'test-pipeline-version-desc',
+          'test-pipeline-desc',
+          'test-default-version-desc',
+          false,
+        )}
+      />,
+    );
+
+    expect(screen.getByText('test-pipeline-desc'));
+    expect(screen.getByText('test-default-version-desc'));
+    expect(screen.queryByText('test-pipeline-version-desc')).toBeNull();
+    expect(screen.getByText('Default Version Description'));
+  });
+
+  it('hides pipeline description when not set with default version', async () => {
```

Comment:
Actually, it can be empty. Although the UI shows it as a required field, the UI does not block you when you do not add a description.

but I agree with the conclusion, we should always show "Pipeline Description", when it's empty, we can show an empty notice like "empty description".